### PR TITLE
fix(components/Datalist): select suggestion value from typed title

### PR DESCRIPTION
**Pull request by jare-talend**
**What is the problem this PR is trying to solve?**
On the datalist, after one search, the input is erased.
ui_datalist_before

**What is the chosen solution to this problem?**

On the datalist, after one search, we kept the value only if a value matching with one suggestion
ui_datalist_after

This behavior is checked with marc.

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

**Comment by frassinier** [comment]
It's funny since it's exactly what a datalist does not do?
> https://developer.mozilla.org/en-US/docs/Web/HTML/Element/datalist

**Comment by jare-talend** [comment]
> It's funny since it's exactly what a datalist does not do?
> 
> > https://developer.mozilla.org/en-US/docs/Web/HTML/Element/datalist

yes, it doesn't. also it made less thing. It accepts free type, no validation control, ...

**Comment by frassinier** [comment]
@jare-talend so it's more a `<select />` behavior that you want to apply to a `<datalist />`
I personally not convinced at all by this feature which it's pretty disturbing

**Comment by jare-talend** [comment]
> @jare-talend so it's more a `<select />` behavior that you want to apply to a `<datalist />`
> I'm personally not convinced at all by this feature which it's pretty disturbing

search and select, that is wanted. @jsomsanith set a no go on that PR. So, I wait reviews with UX team to see the wished behavior.

File: packages/components/src/Datalist/Datalist.component.js [hunk 1]
```diff
@@ -64,7 +64,18 @@ class Datalist extends Component {
 		const { value, previousValue } = this.state;
 
 		if (value !== previousValue) {
-			this.updateValue(event, value, true);
+			let newValue = value;
+			if (this.props.restricted && !this.props.multiSection && this.state.suggestions && value) {
+				const hasMatchingSuggestion = this.state.suggestions.find(
```
Comment (jsomsanith) [inline packages/components/src/Datalist/Datalist.component.js]: You should base your search on titleMap props, not suggestions. 
Here it means that it doesn't take the suggestion if you close suggestions.

File: packages/components/src/Datalist/Datalist.component.test.js [hunk 2]
```diff
@@ -214,10 +214,39 @@ describe('Datalist component', () => {
...
+				multiSection={false}
+				errorMessage={'This should be correct'}
+				onChange={onChange}
+				{...props}
+				value={'foo'}
+				restricted
+			/>,
+		);
+		const input = wrapper.find('input').at(0);
+		input.simulate('change', { target: { value: 'foobar' } });
```
Comment (jsomsanith) [inline packages/components/src/Datalist/Datalist.component.test.js]: Can you change you use case to have title != value please ?
So we know that in you case, with `{title: 'lol', value: 'mdr'}`, typing `lol` propagates `mdr`

**Pull request merged**